src/control/server/config/server.go

```diff
@@ -347,6 +313,10 @@ func (cfg *Server) Load() error {
 	// propagate top-level settings to server configs
 	for i := range cfg.Engines {
 		cfg.updateServerConfig(&cfg.Engines[i])
+		cfg.Engines[i].
+			WithValidateProvider(netdetect.ValidateProviderConfig).
+			WithGetIfaceNumaNode(netdetect.GetIfaceNumaNode).
+			WithGetNetDevCls(netdetect.GetDeviceClass)
```

Comment:
Is this a necessary move? There will be conflicts with #7686 regardless, but I'd like to keep it down to only the necessary ones.

I know I said before that your PR should go in first (and maybe it still should) but it may be easier to solve the problem you're looking to solve with the changes in my PR.

Reply:
this is necessary to give any "real" config the right methods to do the necessary lookups after parsing